libdnf/module/ModulePackage.cpp

```diff
@@ -0,0 +1,358 @@
+}
+
+/**
+ * @brief Return module $version converted to string.
+ *
+ * @return std::string
+ */
+std::string ModulePackage::getVersion() const
+{
+    return std::to_string(metadata->getVersion());
+}
+
+/**
+ * @brief Return module $context.
+ *
+ * @return std::string
+ */
+std::string ModulePackage::getContext() const
+{
+    return metadata->getContext();
+}
+
+/**
+ * @brief Return module $arch.
+ *
+ * @return std::string
+ */
+const char * ModulePackage::getArchitecture() const
+{
+    return metadata->getArchitecture();
+}
+
+/**
+ * @brief Return module $name:$stream:$version:$context:$arch.
+ *
+ * @return std::string
+ */
+std::string ModulePackage::getFullIdentifier() const
+{
+    std::ostringstream ss;
+    ss << getName() << ":" << getStream() << ":" << getVersion() << ":" << getContext() << ":"
+       << getArchitecture();
+    return ss.str();
+}
+
+/**
+ * @brief Return module $summary.
+ *
+ * @return std::string
+ */
+std::string ModulePackage::getSummary() const
+{
+    return metadata->getSummary();
+}
+
+/**
+ * @brief Return module $description.
+ *
+ * @return std::string
+ */
+std::string ModulePackage::getDescription() const
+{
+    return metadata->getDescription();
+}
+
+/**
+ * @brief Return list of RPM NEVRAs in a module.
+ *
+ * @return std::vector<std::string>
+ */
+std::vector<std::string> ModulePackage::getArtifacts() const
+{
+    return metadata->getArtifacts();
+}
+
+/**
+ * @brief Return a module profile by name.
+ *
+ * @return std::shared_ptr<ModuleProfile>
+ */
+std::shared_ptr<Profile> ModulePackage::getProfile(const std::string &name) const
+{
+    return metadata->getProfile(name);
+}
+
+/**
+ * @brief Return list of ModuleProfiles.
+ *
+ * @return std::vector<std::shared_ptr<ModuleProfile>>
+ */
+std::vector<std::shared_ptr<ModuleProfile>> ModulePackage::getProfiles() const
+{
+    return metadata->getProfiles();
+}
+
+/**
+ * @brief Return list of ModuleDependencies.
+ *
+ * @return std::vector<std::shared_ptr<ModuleDependencies>>
+ */
+std::vector<std::shared_ptr<ModuleDependencies>> ModulePackage::getModuleDependencies() const
+{
+    return metadata->getDependencies();
+}
+
+/**
+ * @brief Is a ModulePackage part of an enabled stream?
+ *
+ * @return bool
+ */
+bool ModulePackage::isEnabled()
+{
+    return state == ModuleState::ENABLED;
+}
+
+/**
+ * @brief Mark ModulePackage as part of an enabled stream.
+ */
+void ModulePackage::enable()
+{
+    state = ModuleState::ENABLED;
+}
+
+/**
+ * @brief Add conflict with a module stream represented as a ModulePackage.
+ */
+void ModulePackage::addStreamConflict(const std::shared_ptr<ModulePackage> &package)
+{
+    std::ostringstream ss;
+    Solvable *solvable = pool_id2solvable(pool, id);
+
+    ss << "module(" + package->getNameStream() + ")";
+    auto depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_CONFLICTS, depId, -1);
+}
+
+static std::string getPlatformStream(const std::string &osReleasePath)
```

Comment:
Detection of platform will fail:
1. For empty installroot
2. Anaconda
3. System-upgrade
The solution should be delivered by Modularity team.